api/v1beta1/azuremanagedcontrolplane_types.go

```diff
@@ -221,6 +265,23 @@ type AzureManagedControlPlaneSpec struct {
 	// DisableLocalAccounts disables getting static credentials for this cluster when set. Expected to only be used for AAD clusters.
 	// +optional
 	DisableLocalAccounts *bool `json:"disableLocalAccounts,omitempty"`
+
+	// AutoUpgradeProfile - Profile of auto upgrade configuration.
+	// +optional
+	AutoUpgradeProfile *ManagedClusterAutoUpgradeProfile `json:"autoUpgradeProfile,omitempty"`
+}
+
+// ManagedClusterAutoUpgradeProfile - Auto upgrade profile for a managed cluster.
+type ManagedClusterAutoUpgradeProfile struct {
+	// NodeOSUpgradeChannel is a manner in which the OS on your nodes is updated. The default is NodeImage. Possible values include: NodeImage,Unmanaged,None
```

Comment:
not defaulting it any where actually.

Reply:
Not sure if we have to default it, because i think if we don't set a value as well it would be NodeImage.

Reply:
> Not sure if we have to default it, because i think if we don't set a value as well it would be NodeImage.

I see. You mean this gets set to `NodeImage` by the Azure API?

I wonder if we could add an annotation to default this field to `NodeImage`? I ask since the comment says `The default is NodeImage` It might puzzle the user if it doesn't reflect on the CAPZ spec when left uninitialized. What do you think?

Reply:
I did not understand what you are trying to say, by adding an annotation how can annotation default the field ?

Reply:
added defaulting for NodeOsImage